src/builder.rs

```diff
@@ -575,6 +584,14 @@ async fn send_all_socket(
     Ok(())
 }
 
+fn connect_to_socket(path: &PathBuf, socket: &UnixDatagram, socket_connected: &mut bool) {
+    if let Err(e) = socket.connect(path) {
+        error!("error connecting to socket {path:?}: {e}");
+    } else {
+        *socket_connected = true;
+    }
+}
+
```

Comment:
Aesthetically I'd prefer this function to be defined within the SocketGateway path, possibly even as a closure so it could take no arguments?